### DISCUSSION TITLE
C++ how to reverse a dictionary?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#141390] [Programming Help]
### Body

```cpp
#include <map>
#include <iostream>
#include <string>
class IL{
public:
    std::map<std::string, std::string>x64 = {
        {"ACCUMULATOR", "RAX"},
        {"BASE", "RBX"},
        {"COUNTER", "RCX"},
        {"STACK", "RSP"},
        {"COMPARE", "CMP"},
        {"JUMP", "JMP"},
        {"JUMPLESS", "JL"},
        {"JUMPEQUAL", "JE"},
        {"JUMPIFNOTEQUAL", "JNE"},
        {"JUMPIFGREATER", "JG"},
        {"JUMPIFNOTGREATER", "JNG"},
    };
    inline std::string get_x64instruction(std::string instruction){
        if(this->x64.find(instruction) != this->x64.end()){
            return this->x64[instruction] + "\n";
        }else{
            return std::to_string(toascii(false));
        }
        return "\n";
    }
    inline std::string reverse_x64_dictionary(){
        return "I hate C++";
    }
};
int main(){
    IL il;
    std::cout << il.reverse_x64_dictionary();
}
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mostafalarki1368** [comment]
> ### Body
> 
> ```cpp
> #include <map>
> #include <iostream>
> #include <string>
> class IL{
> public:
>     std::map<std::string, std::string>x64 = {
>         {"ACCUMULATOR", "RAX"},
>         {"BASE", "RBX"},
>         {"COUNTER", "RCX"},
>         {"STACK", "RSP"},
>         {"COMPARE", "CMP"},
>         {"JUMP", "JMP"},
>         {"JUMPLESS", "JL"},
>         {"JUMPEQUAL", "JE"},
>         {"JUMPIFNOTEQUAL", "JNE"},
>         {"JUMPIFGREATER", "JG"},
>         {"JUMPIFNOTGREATER", "JNG"},
>     };
>     inline std::string get_x64instruction(std::string instruction){
>         if(this->x64.find(instruction) != this->x64.end()){
>             return this->x64[instruction] + "\n";
>         }else{
>             return std::to_string(toascii(false));
>         }
>         return "\n";
>     }
>     inline std::string reverse_x64_dictionary(){
>         return "I hate C++";
>     }
> };
> int main(){
>     IL il;
>     std::cout << il.reverse_x64_dictionary();
> }
> ```
> 
> ### Guidelines
> 
> - [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mostafalarkii** [comment]
Admin ***@***.***

در تاریخ دوشنبه 14 اکتبر 2024،‏ 5:05 MyBrain=NULL (Ethan James John
Mullally) ***@***.***> نوشت:

**Comment by Vpngod1** [comment]
Bs
